app/creator/[creatorId]/page.tsx

```diff
@@ -2,26 +2,35 @@ import { FRAME_INPUT_PLACEHOLDER, VERCEL_URL } from '@/lib/consts';
 import LeaderboardPage from '@/components/LeaderboardPage/LeaderboardPage';
 import { FrameMetadata } from '@coinbase/onchainkit';
 import getButtons from '@/lib/getButtons';
+import { isAddress } from 'viem';
+import { ethPublicClient } from '@/lib/publicClient';
 
-const Page = ({ params }: { params: { creatorId: string } }) => (
-  <div>
-    <FrameMetadata
-      ogTitle="data muse"
-      ogDescription="data muse"
-      buttons={getButtons()}
-      image={{
-        src: `${VERCEL_URL}/api/leaderboard?creator=${params.creatorId}`,
-        aspectRatio: '1:1',
-      }}
-      input={{
-        text: FRAME_INPUT_PLACEHOLDER,
-      }}
-      postUrl={`${VERCEL_URL}/api/frame`}
-    />
-    <meta property="of:accepts:xmtp" content="2024-02-01" />
+const Page = async ({ params }: { params: { creatorId: string } }) => {
+  const getProperAddress = async (creator: string) => {
+    return !isAddress(creator)
+      ? await ethPublicClient.getEnsAddress({ name: creator })
+      : params.creatorId;
+  };
```

Comment:
I would move this to a standalone lib
- `/lib/getProperAddress.tsx`

Reply:
done moved to new file